src/index.js

```diff
@@ -0,0 +1,157 @@
+    mealsCount = data.length;
+  }
+  return mealsCount;
+};
+
+// Involvement API functions
+const getInvolvementData = async () => {
+  const response = await getLikes();
+  return response;
+};
+
+const addLike = async (event) => {
+  const heart = event.target;
+  const response = await createLike(heart.id);
+  if (response === true) {
+    heart.src = redHeart;
+    const container = heart.parentElement;
+    const hidden = container.childNodes[0];
+    const previous = hidden.value;
+    const actual = parseInt(previous, 10) + 1;
+    hidden.value = actual;
+    const label = container.childNodes[2];
+    label.innerText = `${actual} likes`;
+  }
+};
+
+const getCommentsById = async (itemId) => {
+  const comments = await getComments(itemId);
+  const countComments = (comments === undefined) ? 0 : comments.length;
+  return { countComments, comments };
+};
+
+const getReservationsById = async (itemId) => {
+  const reservations = await getReservations(itemId);
+  const countReservations = (reservations === undefined) ? 0 : reservations.length;
+  return { countReservations, reservations };
+};
+const submitComment = async (e) => {
+  e.preventDefault();
+  const inputName = document.getElementById('name');
+  const inputInsights = document.getElementById('insights');
+  const id = document.getElementsByName('mealId')[0].value;
+  addComment(id, inputName.value, inputInsights.value);
+  const comments = document.getElementsByClassName('comments')[0];
+  const liComment = document.createElement('li');
+  const date = new Date();
+  liComment.innerHTML = `${date.getFullYear()}-${date.getMonth()}-${date.getDay()} ${inputName.value}: ${inputInsights.value}`;
+  comments.appendChild(liComment);
+  const countComments = document.getElementsByClassName('countComments')[0];
+  countComments.innerHTML = `Comments(${comments.childNodes.length})`;
+  const form = document.getElementsByTagName('form')[0];
+  form.reset();
+};
+const closePopup = () => {
+  const popupContainer = document.getElementsByClassName('popupContainer')[0];
+  popupContainer.style.display = 'none';
+  document.removeChild(popupContainer);
+};
+const showCommentsInfo = async (event) => {
+  const button = event.target;
+  const mealId = button.id;
+  const mealInfo = await getMealInfo(mealId);
+  if (mealInfo !== undefined) {
+    const meal = Object.values(mealInfo.meals)[0];
+
+    const { countComments, comments } = await getCommentsById(mealId);
+    let commentsSection = '';
+    if (countComments > 0) {
+      comments.forEach((comment) => {
+        commentsSection += `<li>${comment.creation_date} ${comment.username}: ${comment.comment}</li>`;
+      });
+    }
+    const popupContainer = document.createElement('div');
+    popupContainer.classList.add('popupContainer');
+    const popup = document.createElement('div');
+    popup.classList.add('popup');
+    popup.innerHTML = `<div class='heading'> <a class='close' href='#'><img src='${close}' alt='close icon'></a></div>
```

Comment:
- [ ] [OPTIONAL] Great job with the modal comment section, however, let's remove the excess space we have up here.
![Screenshot from 2022-04-23 23-46-26](https://user-images.githubusercontent.com/58692590/164948899-96598d68-b390-4a4e-b1b8-39fb5bc4dca1.png)